tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/statements/compoundassignment/CompoundAssignmentTest.java

```diff
@@ -321,6 +322,16 @@ public void testCompoundAssignmentAdditionStructElements() {
         Assert.assertEquals(((BInteger) returns[0]).intValue(), 500);
     }
 
+    @Test(description = "Test compound assignment with addition.")
+    public void testCompoundAssignmentOfXmlAndString() {
+        BValue[] returns = BRunUtil.invoke(result, "xmlCompoundAssignment");
+        Assert.assertTrue(returns[0] instanceof BXMLSequence);
+        BXMLSequence sequence = (BXMLSequence) returns[0];
+        Assert.assertEquals(sequence.size(), 4);
```

Comment:
Just to clarify based on this example, according to the spec, "if one operand is a string and one is xml, the string is treated as if it were an xml sequence with one character item for each code point in the string".

So when we are adding "hah" and "blah", is it not like adding two sequences of lengths 3 and 4 respectively?

Reply:
Yes you are correct, there is a bug in BXMLSequence.size() method. Will fix that.